### PR TITLE
8311902: Concurrency regression in the PBKDF2 key impl of SunJCE provider

### PR DESCRIPTION
**Pull request by valeriepeng**
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8311902](https://bugs.openjdk.org/browse/JDK-8311902): Concurrency regression in the PBKDF2 key impl of SunJCE provider (**Bug** - P2)

### Reviewers
 * [Sean Mullan](https://openjdk.org/census#mullan) (@seanjmullan - **Reviewer**)
 * [Anthony Scarpino](https://openjdk.org/census#ascarpino) (@ascarpino - **Reviewer**)
 * [Xue-Lei Andrew Fan](https://openjdk.org/census#xuelei) (@XueleiFan - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk21.git pull/139/head:pull/139` \
`$ git checkout pull/139`

Update a local copy of the PR: \
`$ git checkout pull/139` \
`$ git pull https://git.openjdk.org/jdk21.git pull/139/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 139`

View PR using the GUI difftool: \
`$ git pr show -t 139`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk21/pull/139.diff">https://git.openjdk.org/jdk21/pull/139.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk21/pull/139#issuecomment-1642517051)

### CODE REVIEW

**Review by seanjmullan — APPROVED** [review]

**Review by ascarpino — APPROVED** [review]

File: src/java.base/share/classes/com/sun/crypto/provider/PBKDF2KeyImpl.java [hunk 1]
```diff
@@ -237,30 +248,45 @@ public String getFormat() {
...
+            }
 
-        if (!(obj instanceof SecretKey))
-            return false;
+            if (!(obj instanceof SecretKey)) {
+                return false;
+            }
 
-        SecretKey that = (SecretKey) obj;
+            SecretKey that = (SecretKey) obj;
```
Comment (XueleiFan) [inline src/java.base/share/classes/com/sun/crypto/provider/PBKDF2KeyImpl.java:274]: is there a tab space?

**Review by XueleiFan — APPROVED** [review]

File: src/java.base/share/classes/com/sun/crypto/provider/PBKDF2KeyImpl.java (hunk 1 above)
Comment (valeriepeng) [reply src/java.base/share/classes/com/sun/crypto/provider/PBKDF2KeyImpl.java:274]: I didn't find a tab space.

**Pull request closed without merging**